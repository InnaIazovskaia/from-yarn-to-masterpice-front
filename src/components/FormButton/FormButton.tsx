import React from "react";
import { Button } from "@chakra-ui/react";

type FormButtonProps = {
  text: string;
  isdDisabled: boolean;
};

const FormButton = ({ text, isdDisabled }: FormButtonProps): JSX.Element => (
  <Button
    type="submit"
    size="lg"
    w="full"
    variant="formButton"
    isDisabled={isdDisabled}
  >
    {text}
  </Button>
);

export default FormButton;
