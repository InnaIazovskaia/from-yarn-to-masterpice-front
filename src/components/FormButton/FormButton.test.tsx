import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";
import FormButton from "./FormButton";

describe("Given a FormButton component", () => {
  afterEach(cleanup);

  const text = "Login";
  const isDisabled = true;

  describe("When it receives 'Login' as text", () => {
    it("Should render button with text 'Login'", () => {
      render(<FormButton text={text} isDisabled={isDisabled} />);

      const button = screen.getByRole("button", { name: text });

      expect(button).toBeInTheDocument();
    });
  });

  describe("When it receives true as isDisabled", () => {
    it("Should be disabled", () => {
      render(<FormButton text={text} isDisabled={isDisabled} />);

      const button = screen.getByRole("button");

      expect(button).toBeDisabled();
    });
  });

  describe("When it's rendered", () => {
    it("Should always match with snapshot", () => {
      const { container } = render(
        <FormButton text={text} isDisabled={isDisabled} />
      );

      expect(container).toMatchSnapshot();
    });
  });
});
