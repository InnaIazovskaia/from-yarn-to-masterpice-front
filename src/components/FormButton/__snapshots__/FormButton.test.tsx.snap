// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given a FormButton component > When it's rendered > Should always match with snapshot 1`] = `
<div>
  <button
    class="chakra-button css-l01cow"
    disabled=""
    type="submit"
  >
    Login
  </button>
</div>
`;
